packages/plugin-ext/src/plugin/tree/tree-views.ts

```diff
@@ -252,8 +252,20 @@ class TreeViewExtImpl<T> implements Disposable {
             dragMimeTypes, dropMimeTypes
         });
         this.toDispose.push(Disposable.create(() => this.proxy.$unregisterTreeDataProvider(treeViewId)));
-        options.treeDataProvider.onDidChangeTreeData?.(() => {
-            this.pendingRefresh = proxy.$refresh(treeViewId);
+        options.treeDataProvider.onDidChangeTreeData?.(elements => {
+            const ids = [];
+            elements = elements || [];
+            if (!Array.isArray(elements)) {
+                elements = [elements];
+            }
+            for (const element of elements) {
+                for (const node of this.nodes.values()) {
+                    if (node.value === element) {
+                        ids.push(node.id);
+                    }
+                }
+            }
```

Comment:
Suggestion: I'm not sure how bad this can get in reality, but this is `O(n * m)` behavior. We can cut this down to `O(n + m)`.
```suggestion
            const set = new Set<T>();
            for (const element of elements) {
                set.add(element);
            }
            for (const node of this.nodes.values()) {
                if (set.has(node.value)) {
                    ids.push(node.id);
                }
            }
```